### DISCUSSION TITLE
dependencyUpdates task not finding any updates

### DISCUSSION THREAD

**Discussion by ColtonIdle** [ben-manes/gradle-versions-plugin#795] [General]
Long time user of this plugin. recently converted my app to use version catalogs. it seems like after that change i have lost the ability to use this plugin.

i tried to create a repro project, but in the simple apps everything works fine with version catalogs + this plugin.

my question is... is there any ideas you have to help me debug why this plugin cant find any updates? it doesn't seem like the plugin never scans my app module at all. i wonder if a convention plugin i have is somehow overriding my root build.gradle file or somethign so its not picking up the tasks?

**Comment by ben-manes** [comment]
It works in [Caffeine](https://github.com/ben-manes/caffeine). Are you using precompiled script plugins to add your dependencies? That acts like a composite build so you have to also run it there directly. I think buildSrc might be similar.

**Reply by ColtonIdle** [reply]
I'm using a setup that is nearly identical to nowinandroid https://github.com/android/nowinandroid in terms of convention plugins. im not using buildSrc or anything.

any ideas on how to debug this to get more info on why it's not finding deps?

For example. the output is

```
------------------------------------------------------------
: Project Dependency Updates (report to plain text file)
------------------------------------------------------------

No dependencies found.

```

but if I run ./gradlew app:dependencies i do get a huge printout of deps

**Reply by ben-manes** [reply]
Convention plugins are precompiled script plugins. The build-logic has its own build and includeBuild creates a composit build for the plugins. That means your plugin dependencies look as if the are transient dependencies which this plugin doesn’t report. In Caffeine I run it on both builds.

A workaround might be to programmatically add all dependencies to a single, custom configuration. Then when that is resolved it would report it.

**Reply by ColtonIdle** [reply]
Also, this is the output if I add `    id("com.github.ben-manes.versions")` to my app build.gradle.kts plugin block and call app:dependencyUpdates

> Task :app:dependencyUpdates

------------------------------------------------------------
:app Project Dependency Updates (report to plain text file)
------------------------------------------------------------

No dependencies found.

**Reply by ColtonIdle** [reply]
"Convention plugins are precompiled script plugins. "
I did not know! TIL. Weird that NowInAndroid repo works fine.

**Reply by ColtonIdle** [reply]
> A workaround might be to programmatically add all dependencies to a single, custom configuration. Then when that is resolved it would report it.

🤔

**Reply by ColtonIdle** [reply]
> That acts like a composite build so you have to also run it there directly

How would I do that? Sorry for the noob question

**Reply by ben-manes** [reply]
It looks like most of their dependencies are added to the projects rather than through the plugins.

You can iterate over the libs and bundles I think. I do something like that [here](https://github.com/ben-manes/caffeine/blob/09220a0fb3936e0e12b6c86471a2e20e877749b5/gradle/plugins/build.gradle.kts#L30).

**Reply by ColtonIdle** [reply]
hm. most of my depencies are added directly to the :app module too. i feel like something else might actually be broken.

**Reply by ben-manes** [reply]
Oh… you can use —info to see if anything stands out.

There was something about the configuration cache breaking dependency resolution by faking the results when querying a cached configuration. Did you enable that?

**Reply by ColtonIdle** [reply]
on a whim i just removed my use of convention plugins (i have a pretty simple app). and still nothing. 😂 😭 
what the heck could be going on...

**Reply by ColtonIdle** [reply]
ooh. org.gradle.configuration-cache=true. maybe i need to set that to false. lets see!

**Reply by ColtonIdle** [reply]
AHHH THAT WAS IT! Thank you for the tip! I guess I'll disable config cache for now. cheers!

**Reply by ben-manes** [reply]
Here’s the [Gradle explanation](https://github.com/gradle/gradle/issues/24636) and my [debugging](https://github.com/ben-manes/gradle-versions-plugin/issues/666#issuecomment-1489440407). The confusion is that configuration-cache.problems=warn forces it on even though this plugin requests it to be disabled when it’s task is run. Caffeine enables the cache but doesn’t force it on like that.
